jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/test/java/org/jbpm/console/ng/ht/client/editors/taskassignments/TaskAssignmentsPresenterTest.java

```diff
@@ -0,0 +1,153 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.console.ng.ht.client.editors.taskassignments;
+
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.security.shared.api.identity.User;
+import org.jbpm.console.ng.ht.model.TaskSummary;
+import org.jbpm.console.ng.ht.model.events.TaskRefreshedEvent;
+import org.jbpm.console.ng.ht.model.events.TaskSelectionEvent;
+import org.jbpm.console.ng.ht.service.TaskLifeCycleService;
+import org.jbpm.console.ng.ht.service.TaskOperationsService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyString;
+import org.mockito.Mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import org.uberfire.mocks.CallerMock;
+import org.uberfire.mocks.EventSourceMock;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class TaskAssignmentsPresenterTest {
+
+    private static final String CURRENT_USER = "Jan";
+    private static final String OTHER_USER = "OTHER_USER";
+
+    @Mock
+    private TaskAssignmentsPresenter.TaskAssignmentsView viewMock;
+    @Mock
+    private User userMock;
+    @Mock
+    private TaskLifeCycleService lifecycleServiceMock;
+    private Caller<TaskLifeCycleService> lifecycleServiceCallerMock;
+    @Mock
+    private TaskOperationsService operationsServiceMock;
+    private Caller<TaskOperationsService> operationsServiceCallerMock;
+
+    //Thing under test
+    private TaskAssignmentsPresenter presenter;
+
+    @Before
+    public void initMocks() {
+        when(userMock.getIdentifier())
+                .thenReturn(CURRENT_USER);
+
+        lifecycleServiceCallerMock = new CallerMock<TaskLifeCycleService>(lifecycleServiceMock);
+        operationsServiceCallerMock = new CallerMock<TaskOperationsService>(operationsServiceMock);
+
+        presenter = new TaskAssignmentsPresenter(
+                viewMock, userMock,
+                lifecycleServiceCallerMock,
+                operationsServiceCallerMock,
+                new MockEventTaskRefreshed()
+        );
+    }
+
+    @Test
+    public void delegationButtonDisabled_whenDelegationSuccessfull() {
+        presenter.onTaskSelectionEvent(new TaskSelectionEvent(1L));
+        presenter.delegateTask(OTHER_USER);
+
+        verify(viewMock).setDelegateButtonActive(false);
+        verify(viewMock, times(2)).enableDelegateButton(false);
+    }
+
+    @Test
+    public void emptyDelegationUserOrGroup_notAccepted() {
+        presenter.delegateTask("");
+
+        verify(viewMock).setHelpText("DelegationUserInputRequired");
```

Comment:
@Salaboy You're right. I was thinking how to deal with the fact that Constants field in the TaskAssignmentsViewImpl is not initialized. I'll update the PR